backend/research/permissions.py

```diff
@@ -0,0 +1,51 @@
+from rest_framework import permissions
+from users.models import Faculty, Student
+from research.models import ResearchScholar
+
+
+class IsHoD(permissions.BasePermission):
+    """
+    Custom permission to only allow HoD to access.
+    """
+
+    def has_permission(self, request, view):
+        if request.user.groups.filter(name='hod').exists():
+            return True
+        else:
+            return False
+
+
+class IsFaculty(permissions.BasePermission):
+    """
+    Custom permission to only allow faculty to access.
+    """
+
+    def has_permission(self, request, view):
+        if isinstance(request.user, Faculty):
```

Comment:
`isinstance` might not be the best way to go, as we are doing a very redundant check here. Better to use [Model Permissions](https://docs.djangoproject.com/en/2.2/topics/auth/customizing/#custom-permissions) with `request.user.has_perm` and assign these permissions at the Model level. See [DRF Model Perms](https://www.django-rest-framework.org/api-guide/permissions/#djangomodelpermissions).